mustache.js

```diff
@@ -45,6 +45,19 @@
     return obj != null && typeof obj === 'object' && (propName in obj);
   }
 
+  /**
+   * Safe way of detecting whether or not the given thing is a primitive and
+   * whether it has the given property
+   */
+  function primitiveHasOwnProperty (primitive, propName) {  
+    return (
+      typeof primitive !== 'object' 
+      && primitive !== undefined
```

Comment:
Even if it's "redundant" (since `typeof null === 'object'`), I believe the code would be easier to read if this line was `&& primitive != null` (which checks for both `null` and `undefined`), since the fact that `null` is typeof `object` is not very intuitive / clear.

http://2ality.com/2013/10/typeof-null.html